src/compute/src/server.rs

```diff
@@ -325,3 +327,16 @@ pub async fn compute_node_serve(
 
     (join_handle_vec, shutdown_send)
 }
+
+/// Check whether the compute node has enough memory to perform computing tasks. Apart from storage,
+/// it must reserve at least `MIN_COMPUTE_MEMORY_MB` for computing. Otherwise, it is not allowed to
+/// start.
+fn validate_compute_node_memory_config(total_memory_bytes: usize, storage_config: &StorageConfig) {
+    if storage_config.total_memory_limit_mb() << 20 > total_memory_bytes {
+        panic!("The total storage memory capacity exceeds the total memory for the compute node. Please increase the total memory for the compute node or decrease the storage memory capacity in configurations and restart the compute node.");
+    } else if (storage_config.total_memory_limit_mb() + MIN_COMPUTE_MEMORY_MB) << 20
+        >= total_memory_bytes
+    {
+        panic!("No enough memory for computing. Please increase the total memory for the compute node or decrease the storage memory capacity in configurations and restart the compute node.");
```

Comment:
I think forcing a min compute memory requirement is not a suitable behavior, especially when we have no idea the workload of user. I would suggest to do following changes:
1. Rename `MIN_COMPUTE_MEMORY_MB` to `RECOMMEND_MIN_COMPUTE_MEMORY_MB`
2. Print a warning rather panic here.

cc @neverchanje @fuyufjh

Reply:
Will there be any problem if the computing memory is too small? For example, one row in a data chunk may occupy 20B or so. If the computing memory is even less than this value, what behavior of RW is expected?

Reply:
> I think forcing a min compute memory requirement is not a suitable behavior, especially when we have no idea the workload of user. I would suggest to do following changes:
> 
> 1. Rename `MIN_COMPUTE_MEMORY_MB` to `RECOMMEND_MIN_COMPUTE_MEMORY_MB`
> 2. Print a warning rather panic here.
> 
> cc @neverchanje @fuyufjh

In that case we may still need a branch: `if storage_config.total_memory_limit_mb() << 20 >= total_memory_bytes` then `panic!`, because the memory manager must need a positive value of available memory. I think that is a little verbose.

Reply:
Furthermore, "CN total memory = computing memory + storage memory" is somehow too ideal. I would recommend to imagine a "reserved" memory or "system" memory as a margin for

- Allocation overhead (by jemalloc and OS memory management)
- Stack and code segment of prosess
- Network buffer

etc.

500MB is a fair number for these things added up.

Reply:
If we really have to panic here, we should make the message more friendly, e.g. we should print numbers of different parts and the default values so that the user can be clear why it's not enough.